src/web/mjs/connectors/Cuutruyen.mjs

```diff
@@ -0,0 +1,152 @@
+import Connector from '../engine/Connector.mjs';
+import Manga from '../engine/Manga.mjs';
+export default class Cuutruyen extends Connector {
+    constructor() {
+        super();
+        super.id = 'cuutruyen';
+        super.label = 'Cuutruyen';
+        this.tags = ['manga', 'vietnamese'];
+        this.url = 'https://cuutruyen.net';
+        this.api = 'https://kakarot.cuutruyen.net';
+    }
+    async _getMangaFromURI(uri) {
+        const mangaid = uri.href.match(/\/mangas\/([0-9]+)/)[1];
+        const req = new URL('/api/v1/mangas/' + mangaid, this.api);
+        const request = new Request(req, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        return new Manga(this, mangaid, data.data.attributes.name.trim());
+    }
+    async _getMangas() {
+        const uri = new URL('/api/v2/mangas/recently_updated', this.api);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        const pages = data._metadata.total_pages;
+        let mangaList = [];
+        for (let page = 1; page <= pages; page++) {
+            const mangas = await this._getMangasFromPage(page);
+            mangaList.push(...mangas);
+        }
+        return mangaList;
+    }
+    async _getMangasFromPage(page) {
+        const uri = new URL(
+            '/api/v2/mangas/recently_updated?page=' + page,
+            this.api
+        );
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        return data.data.map((element) => {
+            return {
+                id: element.id,
+                title: element.name.trim(),
+            };
+        });
+    }
+    async _getChapters(manga) {
+        const uri = new URL('/api/v2/mangas/' + manga.id + '/chapters', this.api);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        return data.data.map((element) => {
+            return {
+                id: element.id,
+                title: 'Chapter ' + element.number,
+            };
+        });
+    }
+    async _getPages(chapter) {
```

Comment:
If this works, this is not "hakuneko like" . _getPages is supposed to return a list of images url.

If you have to perform some decryption, you have to wrap image urls (and scramble data? ) using_createConnectorURI( {} ) and perform image fetching and decryption in _handleConnectorURI.